src/plugins/identity/contractIdentities.test.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+
+import {node} from "../../core/graphTestUtil";
+import {Graph, NodeAddress} from "../../core/graph";
+import * as Weights from "../../core/weights";
+import {_contractWeightedGraph} from "./contractIdentities";
+
+describe("plugins/identity/contractIdentities", () => {
+  const a = node("a");
+  const b = node("b");
+  const c = node("c");
+  const graph = () => new Graph().addNode(a).addNode(b);
+  const contractions = () => [{old: [a.address, b.address], replacement: c}];
+  const weights = () => {
+    const w = Weights.empty();
+    w.nodeWeights.set(NodeAddress.empty, 3);
+    return w;
+  };
+  describe("_contractWeightedGraph", () => {
+    it("contracts the graph", () => {
+      const wg = {graph: graph(), weights: weights()};
+      const contracted = _contractWeightedGraph(wg, contractions());
+      const expected = graph().contractNodes(contractions());
+      expect(expected.equals(contracted.graph)).toBe(true);
```

Comment:
Nit: prefer a deep equal to see differences if possible, maybe using toJSON for this.

Reply:
expect.equals is deep equals, vs. expect.toBe which checks value equality.

https://jestjs.io/docs/en/expect#toequalvalue

Reply:
We're using `graph.equals` though, not `expect.toEqual`.
The current pattern is `expect(boolean).toBe(true)`.

Reply:
Oh, true. But graph.equals is defined as a proper equality checker for graphs.

Reply:
It is a solid test.
My nit here is that it would be useful feedback during testing to see *how* it differs.
Probably should have formulated it that way. I would like it if a failed test would present me with a diff.

Reply:
Ah, now I understand, thanks.

Not going to change it. (We use this pattern all over the codebase--if we think it's important to provide more useful diffs when checking graph equality, I'd rather try to do it consistently than in just one place.)